pkg/trainer/replicas.go

```diff
@@ -127,10 +135,13 @@ func (s *TFReplicaSet) Create(config *tfv1alpha1.ControllerConfig) error {
 			log.Errorf("Error building PS ConfigMap: %v", err)
 			return err
 		}
-		_, err = s.ClientSet.CoreV1().ConfigMaps(s.Job.job.ObjectMeta.Namespace).Create(cm)
+		createdCM, err := s.ClientSet.CoreV1().ConfigMaps(s.Job.job.ObjectMeta.Namespace).Create(cm)
 		if err != nil {
-			log.Errorf("Error creating PS ConfigMap: %v, %v", cm.ObjectMeta.Name, err)
+			log.Errorf("Error creating PS ConfigMap: %v, %v", createdCM.ObjectMeta.Name, err)
+			s.recorder.Eventf(s.Job.job, v1.EventTypeWarning, FailedCreateReason, "Error creating: %v", err)
 			return err
+		} else {
```

Comment:
nit: we have returned in the previous `if` statement, so `else` condition can be removed here.